### DISCUSSION TITLE
Namespaced identifiers (e.g. PosgreSQL schema)

### DISCUSSION THREAD

**Discussion by cstork** [directus/directus#8410] [Feature Requests]
With this post/discussion I just want to ensure that consideration has been given to namespaced identifiers for tables and other DB objects before finalizing 9.0.

If this can be added in a backward compatible way after 9.0, then that's fine. If not, I'd suggest that this is a paramount feature to consider, because AFAIU the distinguishing feature that Directus has over other similar projects is that it can work with existing DBs. However, any serious PostgreSQL DB will make use of schemas (i.e. namespaces) and, unfortunately, the [search_path workaround](https://github.com/directus/directus/discussions/5395) is limited and risky and does not scale.

**Comment by rijkvanzanten** [comment]
Yup! Also see https://github.com/directus/directus/discussions/3164, https://github.com/directus/directus/discussions/4908

**Comment by rijkvanzanten** [comment]
And yes, I believe this can easily be added in a backwards compatible manner. The default schema is `public`, so as soon as we have a native DB_SCHEMA flag that can just default to `public` and things will continue to work as they do currently 👍🏻

**Reply by cstork** [reply]
Well, what I really mean is, is it possible to use multiple schemas at the same time not just what's defined in DB_SCHEMA?

**Reply by rijkvanzanten** [reply]
Ah I see. No we won't support namespacing collections in distinct schemas within a single instance of Directus at this time 👍🏻

**Reply by cstork** [reply]
Ok, just wanted to make sure that's a conscious decision before finalizing version 9.0. :-)

**Reply by rijkvanzanten** [reply]
Yup! One simple reason for that is the fact that that level of namespacing only exists in a small subset of database vendors we support. Adding that scoping into API endpoints for example would add quite a bit of overhead for a feature that's not super 80/20 and proprietary to postgres/ms sql.

**Reply by cstork** [reply]
Ok, I would have thought many of the cloud-based DBs are based on PG and therefore use namespaces like that. But I'm no expert.

What about allowing a single dot in table names in the future? Is that precluded by your current design?

**Reply by rijkvanzanten** [reply]
> What about allowing a single dot in table names in the future?

That would end up fetching `"public"."fakeSchema.table"`, so I don't think that's what you're looking for

**Reply by cstork** [reply]
That's exactly why I started this discussion, i.e., in order not to paint ourselves in a corner wrt using multiple schemas.  :-)

Why not allow namespaces (ie names with single dots) and interpret the empty DB_SCHEMA such that no prefix is generated? In this case search_path makes sense and would work as expected, i.e., just like in psql.
If DB_SCHEMA is non-empty, the corresponding prefix plus dot is generated.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
